File: 220/ruby/strategy/encoder.rb
class Encoder
  
  def initialize(algorithm)
    @algorithm = algorithm
  end

  def encode(input)
    input.split(' ').map { |word| @algorithm.encode_word(word) }.join(" ") 
  end
  
end

class ReverseEncoding
  
  def encode_word(word)
    word.reverse
  end
  
end

class DoublingEncoding
  
  def encode_word(word)
    word * 2
  end
  
end

File: 220/js/strategy/encoder.js
Encoder = function Encoder(encoding) {
    this.encode = function (string) {
		var words = string.split(" ");
		return words.map(encoding.encodeWord).join(" ");
	};
};

ReverseEncoding = function ReverseEncoding() {
	this.encodeWord = function (word) {
		return word.split("").reverse().join("");
	};
};

DoubleEncoding = function DoubleEncoding() {
	this.encodeWord = function (word) {
		return word + word;
	};
};


